### PR TITLE
mark rmiiop, corba tests optional for run

### PR DESCRIPTION
**Pull request by gurunrao**
mark rmiiop, corba tests optional for run.
details on PR can be found @ https://github.com/eclipse-ee4j/jakartaee-tck/issues/508
Signed-off-by: Gurunandan <gurunandan.rao@oracle.com>

### CODE REVIEW

**Review by RohitKumarJain — APPROVED** [review]
LGTM. For all *.jsp files below Copyright, year needs to be updated.

src/com/sun/ts/tests/integration/session/jspejbjdbc/contentRoot/jsp2ejb.jsp
src/com/sun/ts/tests/integration/sec/propagation/web_to_ejb_noauth.jsp
src/com/sun/ts/tests/integration/sec/propagation/web_to_ejb_auth.jsp
src/com/sun/ts/tests/integration/sec/propagation/ejb_to_ejb.jsp
src/com/sun/ts/tests/integration/entity/jspejbjdbc/contentRoot/jsp2ejb.jsp
src/com/sun/ts/tests/common/vehicle/jsp/contentRoot/jsp_vehicle.jsp

**Comment by gurunrao** [comment]
> LGTM. For all *.jsp files below Copyright, year needs to be updated.
> 
> src/com/sun/ts/tests/integration/session/jspejbjdbc/contentRoot/jsp2ejb.jsp
> src/com/sun/ts/tests/integration/sec/propagation/web_to_ejb_noauth.jsp
> src/com/sun/ts/tests/integration/sec/propagation/web_to_ejb_auth.jsp
> src/com/sun/ts/tests/integration/sec/propagation/ejb_to_ejb.jsp 
> src/com/sun/ts/tests/integration/entity/jspejbjdbc/contentRoot/jsp2ejb.jsp
> src/com/sun/ts/tests/common/vehicle/jsp/contentRoot/jsp_vehicle.jsp

thanks for reviewing, I have updated the copyrights as suggested.

File: Jenkinsfile
Comment (alwin-joseph) [inline Jenkinsfile:198]: I suggest we continue to run the optional tests where these are implemented. We can have a separate run to test the optional techs maybe not that frequently.

Comment (gurunrao) [reply Jenkinsfile:198]: bug states "Make RMIIOP tests configurable to be run or always disabled", hence we have passed by default disable keywords.
Also CORBA, Java IDL, RMI-IIOP are removed technologies - https://jakarta.ee/specifications/platform/9/platform-spec-9-SNAPSHOT.html#a2333
Based on above inputs, disable parameters are passed(have added back rmiiop testsuite in default ts list)

Comment (alwin-joseph) [reply Jenkinsfile:198]: If they are removed tech should we add *_optional for rmiiop

Comment (gurunrao) [reply Jenkinsfile]: rmi/iiop/corba/idl is a unique scenario in which officially tech is removed from the Jakarta EE Platform(like interop), but due to large usage base, tests are retained as optional/configurable/disabled.
We can still treat tests are configurable/optional, but maintain it in disabled state.

File: src/com/sun/ts/lib/harness/keyword.properties
Comment (scottmarlow) [inline src/com/sun/ts/lib/harness/keyword.properties:63]: Should RMI_IIOP is not really optional for Full Platform, should we remove `javaee javaee_optional`?

```
com/sun/ts/tests/connector/localTx/transaction = connector connector_jta_optional rmi_iiop_optional
```
I'm not even sure ^ is right as previously we ran this test if `connector` was specified and still do, is that right?  Would the following be better?

```
com/sun/ts/tests/connector/localTx/transaction = connector_jta_optional rmi_iiop_optional
```
So that this test is only run if either of `connector_jta_optional rmi_iiop_optional` were specified?

File: Jenkinsfile
Comment (alwin-joseph) [reply Jenkinsfile]: IMO the rmiiiop tests should not be run in javaee, javaee_optional, javaee_web_profile_optional or default keywords if they are removed tech…if someone wants to run them use keyword rmiiiop thus they are retained and configurable.

Comment (gurunrao) [reply Jenkinsfile]: javaee_optional, javaee_web_profile_optional have been removed.

**Comment by scottmarlow** [comment]
Point of reference, @dblevins mentioned that we shouldn't need to make keywords.properties changes for tests that only are including javax.rmi classes.

**Comment by gurunrao** [comment]
@dblevins : Is it possible to list down packages as part of RMI/IIOP/CORBA/IDL or share any documentation on the same?

**Review by scottmarlow — APPROVED** [review]

**Comment by scottmarlow** [comment]
The tests which use javax.rmi.PortableRemoteObject are no longer updated by this pull request.

**Comment by scottmarlow** [comment]
Since all of the review comments were addressed, going to merge this now.

**Comment by scottmarlow** [comment]
Will wait until pr is not draft to merge.

**Comment by gurunrao** [comment]
> The tests which use javax.rmi.PortableRemoteObject are no longer updated by this pull request.

done.

**Comment by gurunrao** [comment]
@scottmarlow - as per your input have done following modification:

- connector/localTx/transaction/conSharing2/Client.java + connector/localTx/transaction/conSharing3/Client.java  - import javax.rmi classes that aren't used have been removed.

- EJB tests can still use javax.rmi - I have updated to not mark EJB classes using javax.rmi as rmi_iiop.

**Comment by alwin-joseph** [comment]
As per the current implementation rmi_iiop & corba tests will be run if no keyword(or "all") is entered. That is by default all rmi_iiop & corba tests  will be run along with javaee tests. To skip these tests the user needs to specify !rmi_iiop & !corba in the keyword ; to run only those the user needs to specify rmi_iiop&corba.  Can we confirm that this is same as expected.

This info needs to get into the documentation also once confirmed.

**Comment by scottmarlow** [comment]
> As per the current implementation rmi_iiop & corba tests will be run if no keyword(or "all") is entered. That is by default all rmi_iiop & corba tests will be run along with javaee tests. To skip these tests the user needs to specify !rmi_iiop & !corba in the keyword ; to run only those the user needs to specify rmi_iiop&corba. Can we confirm that this is same as expected.
> 
> This info needs to get into the documentation also once confirmed.

+1 for adding this into the documentation!  There is a separate [issue for updating/improving the keyword documentation](https://github.com/eclipse-ee4j/jakartaee-tck/issues/509) for whomever can work on the doc changes.

**Review by scottmarlow — CHANGES_REQUESTED** [review]
Please see https://github.com/eclipse-ee4j/jakartaee-tck/pull/521#issuecomment-698481573 regarding not running RMIIOP/Corba tests by default.

EJB tests that reference javax.rmi.PortableRemoteObject but do not use RMIIOP/Corba can continue to run by default. 

Sorry about the confusing input yesterday.

File: src/com/sun/ts/lib/harness/keyword.properties
Comment (scottmarlow) [inline src/com/sun/ts/lib/harness/keyword.properties:198]: :+1: 
So these tests are not enabled by default for javaee, instead they are only enabled if rmi-iiop or corba are specified.

File: src/com/sun/ts/lib/harness/keyword.properties
Comment (scottmarlow) [inline src/com/sun/ts/lib/harness/keyword.properties:259]: Which class does this test use for rmi_iiop?  I could be missing something but I only see java.rmi.Remote which can also be used for EJB.

```
grep -ri --no-filename import | sort | uniq
import com.sun.javatest.Status;
import com.sun.ts.lib.harness.*;
import com.sun.ts.lib.porting.*;
import com.sun.ts.lib.util.*;
import com.sun.ts.tests.jaxws.common.*;
import jakarta.ejb.CreateException;
import jakarta.ejb.EJBHome;
import jakarta.ejb.EJBObject;
import jakarta.ejb.SessionBean;
import jakarta.ejb.SessionContext;
import jakarta.ejb.Stateless;
import jakarta.jws.*;
import jakarta.jws.WebService;
import jakarta.xml.ws.Service;
import jakarta.xml.ws.WebServiceException;
import jakarta.xml.ws.WebServiceRef;
import java.rmi.Remote;
import java.rmi.RemoteException;
import java.util.Iterator;
import java.util.Properties;
import javax.naming.InitialContext;
import javax.xml.namespace.QName;
```

File: src/com/sun/ts/tests/integration/sec/propagation/ejb_to_ejb.jsp
Comment (scottmarlow) [inline src/com/sun/ts/tests/integration/sec/propagation/ejb_to_ejb.jsp:25]: :+1:

File: src/com/sun/ts/tests/jacc/util/FetchLogClient.java
Comment (scottmarlow) [inline src/com/sun/ts/tests/jacc/util/FetchLogClient.java:24]: :+1:

File: src/com/sun/ts/lib/harness/keyword.properties
Comment (gurunrao) [reply src/com/sun/ts/lib/harness/keyword.properties]: done

File: bin/xml/ts.top.import.xml
Comment (scottmarlow) [inline bin/xml/ts.top.import.xml:429]: +1

**Review by scottmarlow — APPROVED** [review]

**Pull request merged**
